### PR TITLE
Add dev-steps and gradient clipping

### LINKED ISSUE #21: Exploding gradients problem on the last batch in dev set
In this current version of the code, there is a problem with the `dev-dataset`. After some training epochs,
depending on the `dev-batch-size` argument, the last `batch size` in the validation dataset could become very small
and the loss will be wrongly computed, because it will contain `NaN` values.

This leads to the **exploding gradients** problem where the computed gradients will have big values, hence
the **loss** will increase suddenly.

### PR DESCRIPTION
**Pull request by Iulian277**
Add `dev_dataset.repeat()` and set `clipnorm` argument to 1.0 to avoid the exploding gradients problem.
This PR resolves #21.

### CODE REVIEW

**Comment by cosmoquester** [comment]
Oh very thanks!
